model/src/main/kotlin/PackageReference.kt

```diff
@@ -33,6 +33,14 @@ data class PackageReference(
          */
         val id: Identifier,
 
+        /**
+         * The type of linkage used for the referred package from its dependent package. As most of our supported
+         * [PackageManager]s / languages only support dynamic linking or at least default to it, also use that as the
+         * default value here to not blow up our result files.
+         */
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
```

Comment:
I remember having issues with `NON_DEFAULT` and classes without zero-argument constructor in the past, you might have to use `CUSTOM` in case this does not work.

Reply:
How can I tell? Is it enough to see our tests pass?

Reply:
Yes, that should be enough. IIRC Jackson uses the empty argument constructor to create an instance of the object to find out what the default value is, otherwise it uses the default value of the data type, I guess for an enum the default value is the one with ordinal `0`, so this might coincidentally work. Adding a test could help verifying the intended serialization behaviour.

Reply:
You could try what happens if you switch the order of the constants in `PackageLinkage` out of curiosity :)

Reply:
Our existing tests actually indeed failed, but I found https://github.com/FasterXML/jackson-databind/issues/1346 to fix it.

Reply:
Unfortunately, the above does not seem to work for Kotlin, see https://github.com/FasterXML/jackson-module-kotlin/issues/193. I'll find a work-around.